api/deployments/deployment_controller.go

```diff
@@ -40,6 +40,11 @@ func (dc *deploymentController) GetRoutes() models.Routes {
 			Method:      "GET",
 			HandlerFunc: GetPodLog,
 		},
+		models.Route{
+			Path:        rootPath + "/deployments/{deploymentName}/jobcomponents/{jobComponentName}/scheduledjobs/{scheduledJobName}/logs",
```

Comment:
I think we should use a better Path for logs, perhaps `/applications/{appName}/environments/{envName}/jobcomponents{jobComponentName}/schedulerjobs/{schedulerJobName}/logs`
The handler for this route has a TODO, copied from logs for replicas, and I don't think we should introduce TODOs in new code when we can do it correct. There is no backward compatibility that we need to consider, like for the existing logs endpoint for replicas.
`deploymentName` and `jobComponentName` from the route is not used in the controller or handler code, and the handler reads the `RadixApplication` and enumerates all enviornments in this spec to get enviroment namespace and queries this namespace for the existence of the pod. A lot of extra API calls to the k8s API to check if a pod exist. By using the the suggestet route Path we can extract the environment name from the route. No need to enumerate environments in RadixApplication . Also, we have the `jobComponentName` and can check that the pod defined by `schedulerJobName` actually belongs to the component.

Reply:
fixed